### PR TITLE
Update of Docusaurus to v3 with dependency and syntax enhancements

### PR DESCRIPTION
**Pull request by yataras**
This pull request represents an update to the Docusaurus setup, transitioning from version 2.4.1 to 3.3.2, along with necessary updates to dependencies and syntax to ensure compatibility and functionality enhancements. 

# Thanks for improving Semgrep Docs 😀

### Please ensure

- [ ] A subject matter expert (SME) reviews the content
- [ ] A technical writer reviews the content or PR
- [ ] This change has no security implications or else you have pinged the security team
- [ ] Redirects are added if the PR changes page URLs
- [ ] If you have changed any header tag links (doc/#this-kind-of-anchor), update all instances of that link

### CODE REVIEW

File: docs/semgrep-secrets/policies.md [hunk 1]
```diff
@@ -64,7 +64,7 @@ This section defines the columns of the rule entries in the Policies page:
 | Fix rate  | The percentage of findings that are fixed through changes to the code.  |
 | Severity  | The higher the severity, the more critical the issues that a rule detects.      |
 | Confidence  | Indicates confidence of the rule to detect true positives.      |
-| Source  | Indicates the origin of a rule. | <ul><li><strong>Pro:</strong> Authored by Semgrep. Custom:</strong> Rules created within your Semgrep organization. |
+| Source  | Indicates the origin of a rule. |
```
Comment (khorne3) [inline docs/semgrep-secrets/policies.md:67]:
```suggestion
| Source  | Indicates the origin of a rule. <ul><li><strong>Pro:</strong> Authored by Semgrep. Custom:</strong> Rules created within your Semgrep organization. |
```

File: docs/trophy-case.md [hunk 2]
```diff
@@ -30,8 +30,8 @@ Add yours [with a pull request](https://github.com/semgrep/semgrep-docs)!
...
+| [netskrafl.is](https://github.com/mideind/Netskrafl/pull/76)                                                                                                          | [python.flask.security.xss.audit.template-unescaped-with-safe](https://semgrep.dev/r?q=python.flask.security.xss.audit.template-unescaped-with-safe) | The `\| safe` filter from `from_url` in the `userprefs.html` template causes XSS.|
```
Comment (khorne3) [inline docs/trophy-case.md:34]: This looks good.

File: docs/semgrep-secrets/policies.md (hunk 1 above)
Comment (yataras) [reply docs/semgrep-secrets/policies.md]: Done!

**Comment by s-santillan** [comment]
> [CLA assistant check](https://cla-assistant.io/semgrep/semgrep-docs?pullRequest=1542) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/semgrep/semgrep-docs?pullRequest=1542) before we can accept your contribution.You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/semgrep/semgrep-docs?pullRequest=1542) it.

@yataras Could you please sign the CLA? That way we can merge it as soon as it's ready :)

File: docs/semgrep-code/policies.md [hunk 3]
```diff
@@ -46,10 +46,13 @@ The Policies page consists of a header and three main panes:
         <dd>
             The top header consists of:
             <ul>
-             <li><strong>Policies view drop-down</strong>, which lets you choose between:
-                <ul><li>Grouping rules by vulnerability class</li>
-                <li>No grouping</li>
-                </ul></li>
+             <li>
```
Comment (s-santillan) [inline docs/semgrep-code/policies.md:49]: image

Render looks a little odd, can we fix please?

Comment (yataras) [reply docs/semgrep-code/policies.md]: Sure, fixed.

**Review by khorne3 — APPROVED** [review]

**Review by s-santillan — APPROVED** [review]

**Pull request merged**